### DISCUSSION TITLE
404 Error with OpenAPI

### DISCUSSION THREAD

**Discussion by legalla** [fastapi/fastapi#8992] [Questions] [answered]
Hi all,

New to fastAPI, and coming from Flask, I'm very impressed by the solution.
But using the OpenAPI documentation/UI, I can see in the logs that I have the following issue each time I extend a resource/API :
`INFO:     <IP>:57175 - "GET /%5Bobject%20Module%5D HTTP/1.1" 404 Not Found`
The API is working, but I get this error in the log.

I'm using v0.63.0, on an ubuntu server (18.04)

Thanks a lot for your support.

**Comment by includeamin** [comment]
Hi, Welcome to FastAPI

please follow the [Issue instruction](https://github.com/tiangolo/fastapi/issues/new/choose).
we need more information to help you.
thanks.

**Comment by legalla** [comment]
Hi,

OK, here is what I can add :

It seems this issue appeared when I've implemented the following code (but not sure...) :
```python
class device_import(BaseModel):
    hostname: str
    ipaddress: str
    platform: str
    description: Optional[str] = None
......
@app.post("/system/device", summary="Create a new device in the system", tags=["System"])
async def post_device(device_info: device_import):
........
```
All other routes are using basic structure, with path parameters or Query.

In terms of imports : 
```python
from typing import Optional, Dict
from fastapi import FastAPI, Request, APIRouter, Form, Query, Body
from pydantic import BaseModel, Field
```

This issue appears each time I'm expanding a resource/API in the doc UI (Swagger UI).
Let's say that I click on a GET or a POST object to expand, I will automatically get the following log : 
`
INFO:     <IP>:49702 - "GET /%5Bobject%20Module%5D HTTP/1.1" 404 Not Found
`

As explained, the engine works, and I can launch every request, but I get this error, and I really don't what could be the root cause...

Thanks again,

**Comment by includeamin** [comment]
your route is `POST` and it works fine.
``` python
......
@app.post("/system/device", summary="Create a new device in the system", tags=["System"])
async def post_device(device_info: device_import):
........
```

but your log is about `GET` request.
`Request with GET/HEAD method cannot have body.` [check this out](https://github.com/swagger-api/swagger-ui/issues/2136#issuecomment-216881309)

``` python
from fastapi import Depends

@app.get("/system/device", summary="Create a new device in the system", tags=["System"])
async def post_device(device_info: device_import=Depends()):
    return device_info

```

**Comment by legalla** [comment]
Thanks a lot,

Just checked, and I've got the same issue even if I expand a GET, PUT or DELETE resource.
Let say that I click on this item to expand it, I'll get the log : 
![image](https://user-images.githubusercontent.com/18547804/104301627-2a038c00-54c8-11eb-8fb4-2bd97854127b.png)

`
INFO:    <IP>:51426 - "GET /%5Bobject%20Module%5D HTTP/1.1" 404 Not Found
`

Same thing if I do it on a POST item : 
![image](https://user-images.githubusercontent.com/18547804/104301798-5a4b2a80-54c8-11eb-91bc-793242bdf856.png)

`
INFO:     <IP>:51504 - "GET /%5Bobject%20Module%5D HTTP/1.1" 404 Not Found
`

I have the issue whatever the route and associated method I expand.
But if it has been expanded/loaded 1 time, I will never get the issue for it if I collapse/expand it, until I refresh the page (Which is quite obvious as content as been loaded...)

**Comment by zijam** [comment]
I also get the samme INFO message in my log
```
TRACE:    127.0.0.1:44588 - ASGI [10] Started scope={'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.1'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'client': ('127.0.0.1', 44588), 'scheme': 'http', 'method': 'GET', 'root_path': '', 'path': '/[object Module]', 'raw_path': b'/[object%20Module]', 'query_string': b'', 'headers': '<...>'}
TRACE:    127.0.0.1:44588 - ASGI [10] Send {'type': 'http.response.start', 'status': 404, 'headers': '<...>'}
INFO:     127.0.0.1:44588 - "GET /%5Bobject%20Module%5D HTTP/1.1" 404 Not Found
```

**Comment by mcsheehan** [comment]
I'm having an identical issue to legalla - INFO:     127.0.0.1:58176 - "GET /%5Bobject%20Module%5D HTTP/1.1" 404 Not Found
Whenever I expand any method on the documentation. Also using ubuntu 18.04 - fastapi 0.63.0

**Comment by nickybu** [comment]
I'm also running into the same issue, raised whenever I expand on an endpoint within the Swagger UI.

**Comment by mcsheehan** [comment]
It's easy to reproduce - I have just cloned a git example repo using fast api, it seems to happen on any fast api - when clicking the swagger ui expander. https://developer.mongodb.com/how-to/FARM-Stack-FastAPI-React-MongoDB

**Comment by dxnxk-0** [comment]
Same here.

**Comment by Gatux** [comment]
Same story from swagger-ui as last time #1762 ?

I've got the 404 error with swagger-ui-dist@3.39, so I'll just rollback to swagger-ui-dist@3.38 for now.

**Comment by grayguest** [comment]
> Same story from swagger-ui as last time #1762 ?
> 
> I've got the 404 error with swagger-ui-dist@3.39, so I'll just rollback to swagger-ui-dist@3.38 for now.

Is there a problem with swagger-ui-dist@3.38？if no problem, can you please tell me how to rollback this version, thank you.

**Comment by lakshinav** [comment]
The same thing. Besides when I run my app on localhost it works perfectly. When I run the app from docker container and try to connect to server, running on localhost, I see this error.

**Comment by emir-mx** [comment]
I am having the same error, I think this is just an stetic issue, right?, I am not having any operational issue yet, can you confirm if this could become a pain?

**Comment by ArcLightSlavik** [comment]
Created a pull request with a temporary fix to this (#2656).
Thanks @Gatux for linking the issue with the monkey patch.

**Comment by ArcLightSlavik** [comment]
@emir-mx It's only a docs issue, not a operational issue from any of your code.

**Comment by emir-mx** [comment]
> Created a pull request with a temporary fix to this (#2656).
> Thanks @Gatux for linking the issue with the monkey patch.

Great, It works for me! Thank you

**Comment by mcsheehan** [comment]
Nice thanks @Gatux

**Comment by LackesLab** [comment]
The by @ArcLightSlavik mentioned temp fix works perfectly..

**Comment by athanhat** [comment]
Hi, I started having the same `"GET /%5Bobject%20Module%5D HTTP/1.1" 404 Not Found` issue

```text
INFO:     127.0.0.1:44526 - "GET /api/auth/%5Bobject%20Module%5D HTTP/1.1" 404 Not Found
```
This INFO logger message is triggered the first time I click on any of  the FastAPI Swagger UI buttons in API `/docs` endpoint. Perhaps that will help package authors to trace the problem. 

I'm also using FastAPI v0.63.0 and yes @ArcLightSlavik temp fix works fine.

**Comment by rchojn** [comment]
@athanhat I'm facing the same issue. Any updates on that?

**Comment by jhoelzl** [comment]
#metoo

**Comment by felixdittrich92** [comment]
#same

**Comment by remidinishanth-ntnx** [comment]
`INFO:     127.0.0.1:52860 - "GET /%5Bobject%20Module%5D HTTP/1.1" 404 Not Found` I'm also facing same issue on mac.

**Comment by akshayatbcg** [comment]
I am facing the same issue. In NestJs Swagger UI

**Comment by alexrosenfeld10** [comment]
I'm seeing this as well in Swashbuckle.AspNetcore version 6.0.0. I've opened https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/1993 to try to get more information on the root cause, I know it's only tangentially related but I'm linking here for the sake of anyone else googling this issue.

**Comment by ArcLightSlavik** [comment, marked as answer]
@legalla This was fixed by swagger-ui, could you close the issue?

**Comment by calvo-jp** [comment]
It also returns a `404` response when passing `url` as a path parameter.

```python
from fastapi import APIRouter
from .models import OpenGraphMetaTags

router = APIRouter()

@router.get(
    path='/opengraph-metatags/{url}',
    response_model=OpenGraphMetaTags,
    response_model_by_alias=True,
    response_model_exclude_none=True,
)
async def parse_metatags(url: str):
    ...
```

Above code works if path paremeter is a plain string eg. `some-string`, but not for something like `https://facebook.com`. Terminal logs show that the request url is 

```bash
INFO: 127.0.0.1:59924 - "GET /opengraph-metatags/https%3A//facebook.com HTTP/1.1" 404 Not Found
```

It fails to encode the slashes. On swagger I have the following data, tho
![Screenshot from 2021-11-23 13-59-56](https://user-images.githubusercontent.com/86353187/142977445-3459b898-8dd7-407f-a4b8-95d3f64006b7.png)


I hope this gets fix soon. I really love **FastAPI**

**Comment by 14babarali** [comment]
Why am I getting this "INFO:     127.0.0.1:51291 - "GET /v1/models HTTP/1.1" 404 Not Found - in Fastapi : in my code there so such endpoint

**Reply by YuriiMotov** [reply]
Could you provide [MRE](https://stackoverflow.com/help/minimal-reproducible-example)?
Also, better open new discussion for this
